gcore/gdaldriver.cpp

```diff
@@ -1478,6 +1478,28 @@ bool GDALDriver::CanVectorTranslateFrom(
     return bRet;
 }
 
+bool GDALDriver::HasOpenOption(const char *pszOpenOptionName) const
+{
+    if (pszOpenOptionName == nullptr)
+        return false;
+
+    auto pszOOMd{const_cast<GDALDriver *>(this)->GetMetadataItem(
+        GDAL_DMD_OPENOPTIONLIST)};
+    if (pszOOMd == nullptr)
+        return false;
+
+    // Const cast is safe here since we are only reading the metadata
+    const CPLXMLTreeCloser oXml{CPLParseXMLString(pszOOMd)};
```

Comment:
```suggestion
    // Const cast is safe here since we are only reading the metadata
    auto pszOOMd{const_cast<GDALDriver *>(this)->GetMetadataItem(
        GDAL_DMD_OPENOPTIONLIST)};
    if (pszOOMd == nullptr)
        return false;

    const CPLXMLTreeCloser oXml{CPLParseXMLString(pszOOMd)};
```